### PR TITLE
docs: request and response for payments route

### PR DESCRIPTION
**Pull request by Sangamesh26**
## Type of Change

- [x] Enhancement

## Description

Docs for request and response in the payments routes.

## Motivation and Context

Added docs for the request and responses in payments routes

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -558,12 +558,16 @@ pub enum RoutableConnectors {
...
 #[serde(rename_all = "snake_case")]
 pub enum SupportedWallets {
+    /// Paypal wallet is supported to get the session object
     Paypal,
+    /// Applepay wallet is supported to get the session object
     ApplePay,
+    /// Klarna wallet is supported to get the session object
     Klarna,
+    /// Gpay wallet is supported to get the session object
     Gpay,
```
Comment (SanchithHegde) [inline crates/api_models/src/enums.rs:571]: Most of these wallet names are self-explanatory. Add a doc comment along the lines of "Wallets which support obtaining session object" to the enum instead.

```rust
/// Wallets which support obtaining session object
pub enum SupportedWallets {
    Paypal,
    ApplePay,
    Klarna,
    Gpay,
}
```

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -670,33 +679,44 @@ pub struct PaymentsResponse {
...
     #[serde(rename = "created.gte")]
     pub created_gte: Option<PrimitiveDateTime>,
 }
 
 #[derive(Clone, Debug, serde::Serialize)]
 pub struct PaymentListResponse {
+    // The size of the payment list
     pub size: usize,
+    // The list of payment response
     pub data: Vec<PaymentsResponse>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:720]:
```suggestion
    /// The number of payments included in the list
    pub size: usize,
    /// The list of payments response objects
    pub data: Vec<PaymentsResponse>,
```

You had added regular comments. I'm assuming you meant to add doc comments instead.

File: crates/api_models/src/payments.rs [hunk 3]
```diff
@@ -978,56 +1003,75 @@ pub struct Metadata {
     pub data: serde_json::Value,
 }
 
-#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, ToSchema)]
 pub struct PaymentsSessionRequest {
+    /// The identifier for the payment
     pub payment_id: String,
+    /// This is a 15 minute expiry token which shall be used from the client to authenticate and perform sessions from the SDK
     pub client_secret: String,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1011]:
```suggestion
    /// This is a token which expires after 15 minutes, used from the client to authenticate and create sessions from the SDK
    pub client_secret: String,
```

File: crates/api_models/src/payments.rs [hunk 4]
```diff
@@ -978,56 +1003,75 @@ pub struct Metadata {
...
 
-#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
 pub struct GpayTransactionInfo {
+    /// The country code
     pub country_code: String,
+    /// The currency code
     pub currency_code: String,
+    /// The total price status (Ex: 'FINAL')
     pub total_price_status: String,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1059]: You've used `ex` in some places to denote examples and `Ex` elsewhere. Please try to keep it uniform.

File: crates/api_models/src/payments.rs [hunk 5]
```diff
@@ -1040,56 +1084,87 @@ pub struct GpaySessionTokenData {
...
         merchant_identifier: String,
+        /// The domain name of the merchant which is registered in applepay
         domain_name: String,
+        /// The name to be displayed on applepay button
         display_name: String,
+        /// A string which represents the properties of a payment
         signature: String,
+        /// The identifier for the operational analytics
         operational_analytics_identifier: String,
+        /// The number of retries to get the session repsonse
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1127]:
```suggestion
        /// The number of retries to get the session response
```

File: crates/api_models/src/payments.rs [hunk 6]
```diff
@@ -1040,56 +1084,87 @@ pub struct GpaySessionTokenData {
...
     },
 }
 
-#[derive(Default, Debug, serde::Serialize, Clone)]
+#[derive(Default, Debug, serde::Serialize, Clone, ToSchema)]
 pub struct PaymentsSessionResponse {
+    /// The identifier for the payment
     pub payment_id: String,
+    /// This is a 15 minute expiry token which shall be used from the client to authenticate and perform sessions from the SDK
     pub client_secret: Secret<String, pii::ClientSecret>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1139]: 1. Please update the doc comment here as well.
2. ```suggestion
       #[schema(value_type = String)]
       pub client_secret: Secret<String, pii::ClientSecret>,
   ```

File: crates/api_models/src/payments.rs [hunk 7]
```diff
@@ -1040,56 +1084,87 @@ pub struct GpaySessionTokenData {
...
-#[derive(Default, Debug, serde::Deserialize, serde::Serialize)]
+#[derive(Default, Debug, serde::Deserialize, serde::Serialize, ToSchema)]
 pub struct PaymentsStartRequest {
+    /// Unique identifier for the payment. This ensures impotency for multiple payments
+    /// that have been done by a single merchant. This field is auto generated and is returned in the API response.
     pub payment_id: String,
+    /// The identifier for the Merchant Account.
     pub merchant_id: String,
+    /// The identifier for the payment transaction
     pub txn_id: String,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1168]: Can you rename this field as `attempt_id` instead?

File: crates/api_models/src/payments.rs [hunk 8]
```diff
@@ -978,56 +1003,75 @@ pub struct Metadata {
...
 
-#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, ToSchema)]
 pub struct PaymentsSessionRequest {
+    /// The identifier for the payment
     pub payment_id: String,
+    /// This is a 15 minute expiry token which shall be used from the client to authenticate and perform sessions from the SDK
     pub client_secret: String,
+    /// The list of the supported wallets
     pub wallets: Vec<api_enums::SupportedWallets>,
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:1013]:
```suggestion
    #[schema(value_type = Vec<SupportedWallets>)]
    pub wallets: Vec<api_enums::SupportedWallets>,
```

**Review by SanchithHegde — CHANGES_REQUESTED** [review]
1. Please check the diff for the `openapi/generated.json` file and verify that the required fields are correct as per what we've defined on the request structs.

2. Since this is going to be user/merchant facing public documentation, should we use the complete (and properly capitalized) names of the wallets:
   - `Google Pay` instead of `Gpay`/`gpay`
   - `Apple Pay` instead of `ApplePay`/`Applepay`/`applepay`
   - `PayPal` instead of `Paypal`/`paypal`

I'd like more opinions on 2, preferably from @jarnura, @ashokkjag, @bernard-eugine, @manojradhakrishnan.

**Review by jarnura — APPROVED** [review]
LGTM :)

File: openapi/generated.json [hunk 9]
```diff
@@ -1511,6 +1803,104 @@
...
+            "description": "The identifier for the payment"
+          },
+          "client_secret": {
+            "type": "string",
+            "description": "This is a token which expires after 15 minutes, used from the client to authenticate and create sessions from the SDK"
+          },
+          "session_token": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/SessionToken"
```
Comment (SanchithHegde) [inline openapi/generated.json:1877]: There seems to be no schema for this, please fix it.

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**